### ISSUE TITLE
Transceivers stopped prior to being negotiated are never removed from the set of transcievers

### ISSUE THREAD

**Issue by henbos** [w3c/webrtc-pc#2838]
The spec has this to say about removing stopped transcievers and it happens in response to negotiating:

> - If description is of type "[answer](https://w3c.github.io/webrtc-pc/#dom-rtcsdptype-answer)", then run the following steps:
>
> - - For each transceiver in the connection's [set of transceivers](https://w3c.github.io/webrtc-pc/#transceivers-set) run the following steps:
>
> - - - If transceiver is [stopped](https://w3c.github.io/webrtc-pc/#dfn-stopped), [associated](https://w3c.github.io/webrtc-pc/#dfn-associated) with an m= section and the associated m= section is rejected in connection.[[[CurrentLocalDescription]]](https://w3c.github.io/webrtc-pc/#dfn-currentlocaldescription) or connection.[[[CurrentRemoteDescription]]](https://w3c.github.io/webrtc-pc/#dfn-currentremotedescription), remove the transceiver from the connection's [set of transceivers](https://w3c.github.io/webrtc-pc/#transceivers-set).

Removing transceivers like this is [covered by this WPT](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/web_tests/external/wpt/webrtc/RTCRtpTransceiver-stop.html;l=136;drc=232359e58644a36b534576ec7b65d070c4f75040). The same file also tests that [transceivers stopped prior to being negotiated never show up in the SDP](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/web_tests/external/wpt/webrtc/RTCRtpTransceiver-stop.html;l=22;drc=232359e58644a36b534576ec7b65d070c4f75040) which is also good. (Side-note: this is also consistent with DCs: if the only DC that exists is closed prior to being negotiated, we don't add an m=application line to the SDP. Makes sense!)

But this also makes it possible for a transceiver to end up in "limbo" if stopped prior to being negotiated:
- It will never get an m= section because it is stopped, but it will never be removed from the set of transceivers because it is never formally a rejected m= section.

How do we resolve this issue?
- **Proposal A**: Immediately remove the transceiver from the set of transceivers. Argument: "why need to negotiate something that is not even in the SDP?"
- **Proposal B**: Delay the remove until the next SDP answer is set. Argument: "this is the only place where transceivers are removed today, so let's be consistent with that."

**Comment by henbos** [comment]
Thoughts, my fellow transciever enthusiast @jan-ivar? I'm leaning towards Proposal A but I don't have a strong opinion

**Comment by henbos** [comment]
> Side-note: this is also consistent with DCs: if the only DC that exists is closed prior to being negotiated, we don't add an m=application line to the SDP.

(Actually not sure what the correct behavior is here... but Canary behaves this way as of recently)

**Comment by henbos** [comment]
Moving DC discussion to #2839. Let this issue only talk about transcievers in limbo :)

**Comment by jan-ivar** [comment]
> The same file also tests that [transceivers stopped prior to being negotiated never show up in the SDP](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/web_tests/external/wpt/webrtc/RTCRtpTransceiver-stop.html;l=22;drc=232359e58644a36b534576ec7b65d070c4f75040) which is also good.

- Proposal C: Implement the spec. There is no spec support for that green WPT test.

The spec is quite clear: [stop()](https://w3c.github.io/webrtc-pc/#dom-rtcrtptransceiver-stop): _"Irreversibly marks the transceiver as [stopping](https://w3c.github.io/webrtc-pc/#dfn-stopping-0) ... cause[ing] future calls to [createOffer](https://w3c.github.io/webrtc-pc/#dom-rtcpeerconnection-createoffer) to generate a zero port in the [media description](https://w3c.github.io/webrtc-pc/#dfn-media-description) for the corresponding transceiver, as defined in [[RFC8829](https://w3c.github.io/webrtc-pc/#bib-rfc8829)] ([section 4.2.1.](https://datatracker.ietf.org/doc/html/rfc8829#section-4.2.1)) (The user agent MUST treat a [stopping](https://w3c.github.io/webrtc-pc/#dfn-stopping-0) transceiver as [stopped](https://w3c.github.io/webrtc-pc/#dfn-stopped) for the purposes of RFC8829 only in this case)."_

**Comment by jan-ivar** [comment]
The WPT may predate our stopping hack.

[Section 5.2.1](https://datatracker.ietf.org/doc/html/rfc8829#section-5.2.1) says _"The next step is to generate "m=" sections, as specified in [[RFC4566](https://datatracker.ietf.org/doc/html/rfc4566)], [Section 5.14](https://www.rfc-editor.org/rfc/rfc4566#section-5.14). An "m=" section is generated for each RtpTransceiver that has been added to the PeerConnection, excluding any stopped RtpTransceivers"_.

This seems to have supported the green WPT before the stopping hack, so it would perhaps be nice to preserve that JSEP logic, and I see no need to break it.

If we added another _"MUST treat a [stopping](https://w3c.github.io/webrtc-pc/#dfn-stopping-0) transceiver as [stopped](https://w3c.github.io/webrtc-pc/#dfn-stopped)"_ JSEP hack here as well, then that might support the green WPT.

But with Proposal A we wouldn't even need that hack, I think, which seems promising.

@docfaraday WDYT?

**Comment by henbos** [comment]
Yep, I think the test was prior to our changes and I don't think our changes intended to change this specifically, only address the offerer tagged footgun.

Considering all browsers already align on the old JSEP behavior I think we should just clarfiy/decide when to remove them from the set of transceivers

**Comment by alvestrand** [comment]
So add to the "stopping" steps something like
"if transceiver is not associated with a media section, move to stopped state and remove from the list of transceivers"?
